### PR TITLE
added FilterSubscription example

### PR DESCRIPTION
**Pull request by RalphSteinhagen**
to illustrate a more advanced notify scheme with clients providing a query-parameter-based filter

### CODE REVIEW

File: concepts/majordomo/FilterSubscription_example.cpp [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    std::atomic<int>               receivedA{ 0 };
+    std::atomic<int>               receivedAB{ 0 };
+    clientContextA.subscribe(URI("mds://127.0.0.1:12345/DeviceName/Acquisition?signalFilter=A"), [&receivedA](const opencmw::mdp::Message &update) {
+        fmt::print("Client('A') received message from service '{}' for endpoint '{}'\n", update.serviceName.str(), update.endpoint.str());
+        receivedA++;
+    });
+    clientContextA.subscribe(URI("mds://127.0.0.1:12345/DeviceName/Acquisition?signalFilter=A,B"), [&receivedAB](const opencmw::mdp::Message &update) {
+        fmt::print("Client('A,B') received message from service '{}' for endpoint '{}'\n", update.serviceName.str(), update.endpoint.str());
+        receivedAB++;
+    });
```
Comment (sfalamic) [inline concepts/majordomo/FilterSubscription_example.cpp:164]: Here we subscribe but the clients actually do not receive data from the worker.

**Review by sfalamic — COMMENTED** [review]
This looks good! Are we expecting that the clients actually receive data from the worker?

File: concepts/majordomo/FilterSubscription_example.cpp (hunk 1 above)
Comment (RalphSteinhagen) [reply concepts/majordomo/FilterSubscription_example.cpp:164]: That's why we make these PR reviews!

@sfalamic thanks for spotting this! Will look into this.

Comment (sfalamic) [reply concepts/majordomo/FilterSubscription_example.cpp]: > Should be fixed now, was an error in the WIP OpenCMW client implementation that suppressed the leading '/' of the subscription.

Thank you for the fix. The OpenCMW clients now receive messages from the worker.

> N.B. this problem is specific to the 'Client' implementation. For the REST client this isn't an issue.

Exactly, for the REST client, it already worked as expected.

**Review by sfalamic — APPROVED** [review]
Looks good now!

**Pull request merged**